### PR TITLE
Fixes in validator CLI when serializing dictionary with tuple keys to JSON

### PR DESCRIPTION
**Pull request by seongjun-park-dl**
### Summary
This PR includes:
- Fixes error in validator CLI when serializing a Python dictionary with tuples as keys into JSON format.
- Refactored class interfaces and error messages
- Fixed an error in validator when input dataset does not have LabelCategories

### How to test
`python -m unittest -v tests/test_validator.py`

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Review by jihyeonyi — APPROVED** [review]

**Review by zhiltsov-max — APPROVED** [review]

**Comment by zhiltsov-max** [comment]
Please check the updates.

File: datumaro/components/validator.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ class _Validator:
 
     Attributes
     ----------
-    task_type : str
-        task type (ie. 'classification', etc.) (default is 'classification')
-    ann_type : AnnotationType
+    task_type : str ot TaskType
```
Comment (jihyeonyi) [inline datumaro/components/validator.py:37]: I think it's a typo. ot->or, right?

File: datumaro/components/validator.py [hunk 2]
```diff
@@ -49,10 +49,11 @@ class _Validator:
         Abstract method that must be implemented in a subclass.
     """
 
-    def __init__(self,
-                task_type='classification',
-                ann_type=AnnotationType.label,
-                far_from_mean_thr=None):
+    def __init__(self, task_type=None, ann_type=None, far_from_mean_thr=None):
+        task_type = parse_str_enum_value(task_type, TaskType)
```
Comment (jihyeonyi) [inline datumaro/components/validator.py:53]: please add the default value as 'classification'

**Comment by seongjun-park-dl** [comment]
Fixed the things @jihyeonyi pointed out in the latest commit. Otherwise, the other updates look good to me!

**Review by jihyeonyi — DISMISSED** [review]

**Comment by seongjun-park-dl** [comment]
@maksimvlasov You changed list(d.items()) to d.items() in one of your commits. I changed it back to list(d.items()). The function won't work with d.items() because the .pop() operation changes the length of d.items() dynamically. Casting it as a list solves the problem because it fixes the initial d.items() as a list.

**Comment by zhiltsov-max** [comment]
Right, thanks.

**Pull request merged**